web-common/src/features/dashboards/show-hide-selectors.ts

```diff
@@ -89,6 +95,18 @@ function createShowHideStore<Item>(
     updateMetricsExplorerByName(metricsViewName, (metricsExplorer) => {
       if (metricsExplorer[visibleFieldInStore].has(key)) {
         metricsExplorer[visibleFieldInStore].delete(key);
+
+        /*
+         * If current leaderboard measure is hidden, set the first
+         * visible measure as the current leaderboard measure
+         */
+        if (
```

Comment:
Move this to `syncMeasures` in `dashboard-store.ts`. This way if the list of measures changes in backend then this code is triggered as well.